contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,563 @@
+        if (_safetyPeriod > 6 hours) revert SafetyPeriodTooLong();
+        generalParameters.withdrawPeriod = _withdrawPeriod;
+        generalParameters.safetyPeriod = _safetyPeriod;
+        emit SetWithdrawSafetyPeriod(_withdrawPeriod, _safetyPeriod);
+    }
+
+    /**
+    * @notice Called by governance to set vesting params for rewarding claim
+    * reporters with rewardToken, for all vaults
+    * @param _duration Duration of the vesting period. Must be less than 180 days.
+    * @param _periods The number of vesting periods. Must be more than 0 and 
+    * less then the vesting duration.
+    */
+    function setHatVestingParams(uint256 _duration, uint256 _periods) external onlyOwner {
+        if (_duration >= 180 days) revert VestingDurationTooLong();
+        if (_periods == 0) revert VestingPeriodsCannotBeZero();
+        if (_duration < _periods) revert VestingDurationSmallerThanPeriods();
+        generalParameters.hatVestingDuration = _duration;
+        generalParameters.hatVestingPeriods = _periods;
+        emit SetHatVestingParams(_duration, _periods);
+    }
+
+    /**
+    * @notice Called by governance to set the timelock delay for setting the
+    * max bounty (the time between setPendingMaxBounty and setMaxBounty)
+    * @param _delay The time period for the delay. Must be at least 2 days.
+    */
+    function setMaxBountyDelay(uint256 _delay)
+    external
+    onlyOwner {
+        if (_delay < 2 days) revert DelayTooShort();
+        generalParameters.setMaxBountyDelay = _delay;
+        emit SetMaxBountyDelay(_delay);
+    }
+
+    /**
+    * @notice Create a new vault
+    * @param _asset The vault's native token
+    * @param _committee The address of the vault's committee 
+    * @param _rewardController The reward controller for the vault
+    * @param _maxBounty The maximum percentage of the vault that can be paid
+    * out as a bounty. Must be between 0 and `HUNDRED_PERCENT`
+    * @param _bountySplit The way to split the bounty between the hacker, 
+    * committee and governance.
+    *   Each entry is a number between 0 and `HUNDRED_PERCENT`.
+    *   Total splits should be equal to `HUNDRED_PERCENT`.
+    *   Bounty larger than 0 must be specified for the hacker (direct or 
+    *   vested in vault's native token).
+    * @param _descriptionHash Hash of the vault description.
+    * @param _bountyVestingParams Vesting params for the bounty
+    *        _bountyVestingParams[0] - vesting duration
+    *        _bountyVestingParams[1] - vesting periods
+    * @param _isPaused Whether to initialize the vault with deposits disabled
+    * @return vault The address of the new vault
+    */
+    function createVault(
+        IERC20 _asset,
+        address _committee,
+        IRewardController _rewardController,
+        uint256 _maxBounty,
+        HATVault.BountySplit memory _bountySplit,
+        string memory _descriptionHash,
+        uint256[2] memory _bountyVestingParams,
```

Comment:
why is this packed and not just two params?

Reply:
That's how it was before, will change that

Reply:
Oh without it I get stack too deep. But maybe if we remove something else like the setting of the description, or the setting of isPaused. What do you think?